aten/src/ATen/CPUGenerator.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <ATen/core/PhiloxRNGEngine.h>
+
+namespace at {
+
+struct CPUGenerator;
```

Comment:
nit: Just drop the `detail` block below the class, then you don't need a forward declaration anymore